verifiers/internal/gha/provenance_forgeable.go

```diff
@@ -9,13 +9,21 @@ import (
 	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/common"
 	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/iface"
 	slsav02 "github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/v0.2"
+	slsav1 "github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/v1.0"
 )
 
 func verifyProvenanceMatchesCertificate(prov iface.Provenance, workflow *WorkflowIdentity) error {
 	// See the generation at https://github.com/npm/cli/blob/latest/workspaces/libnpmpublish/lib/provenance.js.
 	// Verify systemParameters.
-	if err := verifySystemParameters(prov, workflow); err != nil {
-		return err
+	switch typedProv := prov.(type) {
+	case *slsav1.NpmCLIGithubActionsProvenance:
+		if err := verifyNpmCLIGithubActionsV1SystemParameters(typedProv, workflow); err != nil {
```

Comment:
I can't seem to find why this needs a separate path (vs `verifySystemParamters`)